examples/server-side-rendering/src/app/app.component.ts

```diff
@@ -2,10 +2,30 @@ import { Component, Injector, Inject, PLATFORM_ID } from '@angular/core';
 import { isPlatformServer } from '@angular/common';
 import { HttpClient, HttpHeaders } from '@angular/common/http';
 import { TransferState, makeStateKey } from '@angular/platform-browser';
-import {
-  createSSRSearchClient,
-  parseServerRequest,
-} from 'angular-instantsearch';
+import { history } from 'instantsearch.js/es/lib/routers';
+import { simple } from 'instantsearch.js/es/lib/stateMappings';
+import { createSSRSearchClient } from 'angular-instantsearch';
+import * as qs from 'qs';
+import { Routing } from 'instantsearch.js/es/lib/InstantSearch';
+
+// TODO: put this inside the library, maybe inside InstantSearch.js?
+function parseServerRequest(
+  req: { url: string } | undefined,
+  { router, stateMapping }: Routing
+) {
+  if (!req) {
+    return undefined;
+  }
+
+  // this is for faking the browser API used by the router.
+  const location = { search: `?${req.url.split('?')[1]}` };
+  // TODO: this is "private" in IS.js, but since you can change the parseURL
+  // in historyRouter, we need to use it directly
+  // we can't call .read here, since that uses browser globals, unless it can be injected
+  const routeState = router.parseURL({ qsModule: qs, location });
```

Comment:
We don't necessarily need to use InstantSearch.js internals, especially because we already import `qs` here. Transforming the URL in a UI state is the responsibility of the users so it's fine to let them do this:

```ts
qs.parse(new URL(req.url).search.slice(1), { arrayLimit: 99 })
```

If users create a more sophisticated routing system, they'll handle it here. The function `parseServerRequest` is better user-land (like in React InstantSearch), and can be renamed in the example to something more simple.

Reply:
Hmm, but that wouldn’t work with a custom parseURL no?